homeassistant/components/config/config_entries.py

```diff
@@ -224,3 +232,38 @@ class OptionManagerFlowResourceView(FlowManagerResourceView):
 
         # pylint: disable=no-value-for-parameter
         return await super().post(request, flow_id)
+
+
+@websocket_api.require_admin
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {"type": "config_entries/system_options/list", "entry_id": str}
+)
+async def system_options_list(hass, connection, msg):
+    """List all system options for a config entry."""
+    changes = dict(msg)
+    entry_id = changes.pop("entry_id")
+    entry = hass.config_entries.async_get_entry(entry_id)
+
+    if entry:
+        connection.send_result(msg["id"], entry.system_options.disable_new_entities)
+
+
+@websocket_api.require_admin
+@websocket_api.async_response
+@websocket_api.websocket_command(
+    {
+        "type": "config_entries/system_options/update",
+        "entry_id": str,
+        vol.Optional("disable_new_entities"): bool,
+    }
+)
+async def system_options_update(hass, connection, msg):
+    """Update config entry system options."""
+    changes = dict(msg)
+    changes.pop("type")
+    entry_id = changes.pop("entry_id")
+    entry = hass.config_entries.async_get_entry(entry_id)
+
+    if entry and "disable_new_entities" in changes:
+        entry.system_options.disable_new_entities = changes["disable_new_entities"]
```

Comment:
We should pass this to an update method on system_options. That way that object can be responsible for doing callbacks in the future.